include/tvm/relay/op.h

```diff
@@ -74,16 +74,42 @@ class OpNode : public relay::ExprNode {
     v->Visit("support_level", &support_level);
   }
 
+  /*!
+   * \brief Check that if current op is a "primtive operator".
+   * That is the arguments are all type variables, and there is a single
+   * type relation applied to the input and output types.
+   */
+  bool IsPrimitiveOp() const {
+    if (is_primitive_ != -1) return is_primitive_ != 0;
```

Comment:
if (is_primitive == -1) is_primitive_ = this->Is...;
return is_primitive_ != 0;

Reply:
This code path is more frequent(cached) and can likely hint compiler to do so